src/client/mod.rs

```diff
@@ -290,64 +300,98 @@ pub enum SignerError {
 // -----------------------------------------------------------------------------
 // Signer structure
 
-#[derive(PartialEq, Eq, Clone, Debug)]
-pub struct Signer {
-    pub nonce: String,
-    pub timestamp: u64,
-    pub token: String,
-    pub secret: String,
-    pub consumer_key: String,
-    pub consumer_secret: String,
+#[derive(Clone, PartialEq, Eq)]
+pub struct Signer<T = String> {
```

Comment:
What is the idea behind adding a generic type which is defaulting to the current type? To me, this is over-engineered, please keep it simple stupid

Reply:
IMHO this actually makes things easier and also prevents many intermediate allocations. since we most handle static string slices and types with string slice representations (`Method`, `Url`). I pushed things further in #58 by removing the trait altogether and letting the implementation work on string slices always